deeppavlov/core/data/database.py

```diff
@@ -0,0 +1,187 @@
+                self.cursor.execute("SELECT * FROM {}".format(self.tname) +
+                                    " WHERE {}".format(where_expr) +
+                                    " ORDER BY {} {}".format(order_by, order),
+                                    values)
+            else:
+                self.cursor.execute("SELECT * FROM {}".format(self.tname) +
+                                    " WHERE {}".format(where_expr),
+                                    values)
+        return [self._wrap_selection(s) for s in self.cursor.fetchall()]
+
+    def _wrap_selection(self, selection):
+        if not self.keys:
+            self.keys = self._get_keys()
+        return {f: v for f, v in zip(self.keys, selection)}
+
+    def _get_keys(self):
+        self.cursor.execute("PRAGMA table_info({});".format(self.tname))
+        return [info[1] for info in self.cursor]
+
+    def _get_types(self):
+        self.cursor.execute("PRAGMA table_info({});".format(self.tname))
+        return {info[1]: info[2] for info in self.cursor}
+
+    def fit(self, data: List[Dict]) -> None:
+        if not self._check_if_table_exists():
+            self.keys = self.keys or list(set(k for d in data for k in d.keys()))
+            types = ('integer' if type(data[0][k]) == int else 'text' for k in self.keys)
+            self._create_table(self.keys, types)
+        elif not self.keys:
+            self.keys = self._get_keys()
+
+        self._insert_many(data)
+
+    def _create_table(self, keys, types):
+        if any(pk not in keys for pk in self.primary_keys):
+            raise ValueError("Primary keys must be from {}.".format(keys))
+        new_types = ("{} {} primary key".format(k, t) if k in self.primary_keys else
+                     "{} {}".format(k, t)
+                     for k, t in zip(keys, types))
+        self.cursor.execute("CREATE TABLE IF NOT EXISTS {} ({})"
+                            .format(self.tname, ', '.join(new_types)))
+        log.info("Created table with keys {}.".format(self._get_types()))
+
+    def _insert_many(self, data):
+        to_insert = {}
+        to_update = {}
+        for kv in filter(None, data):
+            primary_values = tuple(kv[pk] for pk in self.primary_keys)
+            record = tuple(kv.get(k, self.unknown_value) for k in self.keys)
+            curr_record = self._get_record(primary_values)
+            if curr_record:
+                if primary_values in to_update:
+                    curr_record = to_update[primary_values]
+                if curr_record != record:
+                    to_update[primary_values] = record
+            else:
+                to_insert[primary_values] = record
+
+        if to_insert:
+            fformat = '(' + ','.join(['?'] * len(self.keys)) + ')'
+            self.cursor.executemany("INSERT into {}".format(self.tname) +
+                                    " VALUES {}".format(fformat),
+                                    to_insert.values())
```

Comment:
Why not
```
fformat =  ','.join(['?'] * len(self.keys))
self.cursor.executemany("INSERT into {} VALUES ({})".format(self.tname, fformat), to_insert.values())
```
?

Reply:
Yeah, this way is defiantly better. I'll fix